crates/ruff_linter/src/fix/edits.rs

```diff
@@ -144,6 +144,34 @@ pub(crate) fn make_redundant_alias<'a>(
         .collect()
 }
 
+/// Fix to add the specified imports to the `__all__` export list.
+pub(crate) fn add_to_dunder_all<'a>(
+    names: impl Iterator<Item = &'a str>,
+    expr: &Expr,
+    stylist: &Stylist,
+) -> Vec<Edit> {
+    let (insertion_point, export_prefix_length) = match expr {
+        ast::Expr::List(ast::ExprList { elts, range, .. })
+        | ast::Expr::Tuple(ast::ExprTuple { elts, range, .. }) => (
+            elts.last()
+                .map(|elt| elt.range().end())
+                .unwrap_or(range.end() - "]".text_len()),
```

Comment:
What if it's an unparenthesized tuple 👀 e.g. https://github.com/python/cpython/blob/460546529b0959ce9528dec1c5cd836dcc04ad2c/Lib/asyncio/base_events.py#L54

There's a field on the `ast::ExprTuple` node that you can query to check whether or not the tuple is parenthesized in the original Python source code

Reply:
Oh what, this is horrible, but ok; willfix

Reply:
I think I've resolved this.